Shared/Samples/Set visibility of subtype sublayer/SetVisibilityOfSubtypeSublayerView.Model.swift

```diff
@@ -0,0 +1,164 @@
+        private var subtypeSublayer: SubtypeSublayer!
+        
+        /// The renderer of the subtype feature layer.
+        private var originalRenderer: Renderer!
+        
+        /// A Boolean value indicating if the sample is authenticated.
+        private var isAuthenticated: Bool {
+            !ArcGISEnvironment.authenticationManager.arcGISCredentialStore.credentials.isEmpty
+        }
+        
+        /// The  subtype sublayer's label definition.
+        private var labelDefinition: LabelDefinition = {
+            // Make and stylize the text symbol.
+            let textSymbol = TextSymbol()
+            textSymbol.backgroundColor = .clear
+            textSymbol.outlineColor = .white
+            textSymbol.color = .blue
+            textSymbol.haloColor = .white
+            textSymbol.haloWidth = 2
+            textSymbol.size = 10.5
+            // Make a label definition and adjust its properties.
+            let labelExpression = SimpleLabelExpression(simpleExpression: "[nominalvoltage]")
+            let labelDefinition = LabelDefinition(labelExpression: labelExpression, textSymbol: textSymbol)
+            labelDefinition.placement = .pointAboveRight
+            labelDefinition.usesCodedValues = true
+            return labelDefinition
+        }()
+        
+        // The formatter used to generate strings from scale values.
+        private let scaleFormatter: NumberFormatter = {
+            let numberFormatter = NumberFormatter()
+            numberFormatter.numberStyle = .decimal
+            numberFormatter.maximumFractionDigits = 0
+            return numberFormatter
+        }()
+        
+        /// A Boolean value indicating whether the settings should be presented.
+        @Published var isShowingSettings = false
+        
+        /// A Boolean value indicating whether to show the subtype sublayer.
+        @Published var showsSublayer = true
+        
+        /// A Boolean value indicating whether to show the subtype sublayer's renderer.
+        @Published var showsOriginalRenderer = true
+        
+        /// The current scale of the map.
+        @Published var currentScale: Double = .zero
+        
+        /// The map's current scale value in text.
+        @Published var currentScaleText: String = ""
+        
+        /// The subtype sublayer's minimum scale value in text.
+        @Published var minimumScaleText: String = "Not Set"
+        
+        init() {
+            map.initialViewpoint = .initialViewpoint
+        }
+        
+        deinit {
+            ArcGISEnvironment.authenticationManager.arcGISCredentialStore.removeAll()
+        }
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async throws {
+            try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+            guard isAuthenticated else { return }
```

Comment:
this just stops the setup with no warning or error. Is that what we want?  Should this be a precondition?

Reply:
You actually do'nt need this because if it fails to add on the line above then the method throws and that is propagated and the method exits.